Sources/Core/Extensions/Other/XCTestCase+URLRequest.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  XCTestCase+URLRequest.swift
+//  PovioKit
+//
+//  Created by Gentian Barileva on 09/09/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import XCTest
+
+public extension XCTestCase {
+  func XCTAssertEqualURLRequests(_ lhs: URLRequest?, _ rhs: URLRequest?, file: StaticString = #filePath, line: UInt = #line) {
```

Comment:
Should we also convert arguments to `@autoclosure`, to follow the XCTest design?

Reply:
I tested this out as follows:
```swift
func XCTAssertEqual(_ lhs: @autoclosure() -> URLRequest?, _ rhs: @autoclosure() -> URLRequest?, file: StaticString = #filePath, line: UInt = #line) {
  XCTAssertTrue(lhs() == rhs(), file: file, line: line)
  guard let lhs = lhs(), let rhs = rhs() else { return }
  XCTAssertEqual(lhs.url, rhs.url, "url", file: file, line: line)
  XCTAssertEqual(lhs.httpMethod, rhs.httpMethod, "httpMethod", file: file, line: line)
  XCTAssertEqual(lhs.httpBody ?? Data(), rhs.httpBody ?? Data(), "httpBody", file: file, line: line)
  XCTAssertEqual(lhs.allHTTPHeaderFields ?? [:], rhs.allHTTPHeaderFields ?? [:], "allHTTPHeaderFields", file: file, line: line)
}
```
But since the original XCTAssertEqual is of generic type I think it's being prioritized instead of our function.